scripts/githooks/pre-commit

```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
```

Comment:
Do you mind if I make format work in 2 different modes? At current "blanket" mode is by default, I would like to add the ability to only clang-format individual files.